### PR TITLE
Added search path for proj.db to Context.create().

### PR DESCRIPTION
**Pull request by ggmuelle**
This is useful if you can't set environment variable PROJ_LIB.

### CODE REVIEW

**Comment by desruisseaux** [comment]
Thanks for the patch. The addition of search path seems fine. But should we really include the PROJ headers and binaries in the GitHub repository? Versionning binary files together with source codes is [not recommended](https://robinwinslow.uk/dont-ever-commit-binary-files-to-git). Especially for executables: why only Windows binaries and not the other platforms? Why this specific PROJ version? Are we going to push new binaries at every PROJ  releases? Who will do the maintenance work of keeping those binaries and headers up-to-date?

Would it be possible to modify this pull request so that it addresses only the search path issue, without adding binaries? Instead of binaries, I would suggest to try to improve the `INSTALL` guide. If nevertheless there is a wish from the community to add binaries, then I think it should be a separated discussion not be be taken lightly.

**Comment by ggmuelle** [comment]
Sorry, adding the includes and libraries should be only for internal use. I thought only the search-path-stuff will be part of the pull request because I've created it before adding the includes and libraries. I'll try to recreate the PR.

**Comment by desruisseaux** [comment]
Thanks! No need to recreate a new merge request. A squashing with `git rebase -i` followed by a `git push --force` will work well.

**Comment by ggmuelle** [comment]
The PR is clean now.

File: src/main/cpp/bindings.cpp [hunk 1]
```diff
@@ -786,9 +786,18 @@ JNIEXPORT jobject JNICALL Java_org_osgeo_proj_UnitOfMeasure_create
  * @param  caller  The class from which this method has been invoked.
  * @return The address of the new PJ_CONTEXT structure, or 0 in case of failure.
  */
-JNIEXPORT jlong JNICALL Java_org_osgeo_proj_Context_create(JNIEnv *env, jclass caller) {
+JNIEXPORT jlong JNICALL Java_org_osgeo_proj_Context_create(JNIEnv *env, jclass caller, jstring searchPaths) {
     static_assert(sizeof(PJ_CONTEXT*) <= sizeof(jlong), "Can not store PJ_CONTEXT* in a jlong.");
     PJ_CONTEXT *ctx = proj_context_create();
+	
+	if (searchPaths != NULL) {
```
Comment (desruisseaux) [inline src/main/cpp/bindings.cpp:793]: Would it be possible to replace tabulations by 4 spaces?

**Comment by ggmuelle** [comment]
Done.

**Comment by desruisseaux** [comment]
I added a commit with the addition of documentation and for mutl-directories support. I took the initiative to rename the `proj.data` property as `org.osgeo.proj.data` for consistency with the existing `org.osgeo.proj.maxThreadsPerInstance` property ([javadoc](https://osgeo.github.io/PROJ-JNI/org.osgeo.proj/org/osgeo/proj/package-summary.html#org.osgeo.proj.maxThreadsPerInstance)). Please let me know if this is an issue.

**Comment by ggmuelle** [comment]
I'm fine with that.

**Comment by desruisseaux** [comment]
For the record, a similar functionality was provided in an independent fork [here](https://github.com/interactive-instruments/PROJ-JNI/compare/f6b397b9e0a6da2be01ca37e9a505a432ee0b00b..f993e158ef81f618e691979428f30349645d4e40).

**Pull request merged**